### ISSUE TITLE
default.cache.php file not written correctly

### ISSUE THREAD

**Issue by webandflow** [modxbot/migrate#4728]
webandflow created Redmine issue ID 4728

On several occasions now, I've been happily working away in the 2.1.0-rc4 manager and, following some action (typically saving a resource or element), the server beings returning 500 errors.

One thing I have noticed is that it seems like core/cache/lexion_topics/lexicon/en/default.cache.php get written incorrectly/strangely.  While this file shoudl start with:

<pre>
<?php return array {
</pre>

Instead it starts with this:

<pre>
=> 'The time in this field must be equal to or after {0}',
</pre>

This, I'm guessing is the continuation of the last line in the file:

<pre>
'ext_timemin'
</pre>

For whatever reason, this file gets split down the middle and the second half gets written first.

When this has happened, I've been able to delete this cache file and the manager will continue to work.

**Comment by opengeek** [comment]
opengeek submitted:

Well, I certainly can't reproduce this by just saving Resources and Elements. If you can find more specific steps that make this reproducible, I will review further.

**Comment by webandflow** [comment]
webandflow submitted:

I've been scouring through some logs and trying to reproduce.  Can't seem to figure out any steps to reproduce the problem, but my apache access log may have turned up something interesting.

First, though - it seems like it's not just the default.cache.php file that is affected.  The last time the server returned this error - that file wasn't affected, and I couldn't find one that was.  The problem went away when I re-ran the setup process.

But, my apache logs are showing that every time I've triggered this problem (at least recently), it's started with a POST to /connectors/element/propertyset.php.  Here's a typical string of entries from my log (starting with the bad POST, followed by my attempts to load the manager).

@access.log:3150: 66.195.203.33 - - [13/May/2011:12:53:41 -0400] "POST /connectors/element/propertyset.php HTTP/1.1" 500 - "http://www.thisismydomain.org/manager/index.php?a=19&id=17" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.24 (KHTML, like Gecko) Chrome/11.0.696.68 Safari/534.24"
access.log:3151: 66.195.203.33 - - [13/May/2011:12:53:44 -0400] "GET /manager/index.php?a=19&id=33 HTTP/1.1" 500 - "http://www.thisismydomain.org/manager/index.php?a=19&id=17" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.24 (KHTML, like Gecko) Chrome/11.0.696.68 Safari/534.24"
access.log:3153: 66.195.203.33 - - [13/May/2011:12:53:55 -0400] "GET /manager/index.php?a=19&id=33 HTTP/1.1" 500 - "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.24 (KHTML, like Gecko) Chrome/11.0.696.68 Safari/534.24"
access.log:3155: 66.195.203.33 - - [13/May/2011:12:54:22 -0400] "GET /manager/ HTTP/1.1" 500 - "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.24 (KHTML, like Gecko) Chrome/11.0.696.68 Safari/534.24"
access.log:3156: 66.195.203.33 - - [13/May/2011:12:54:22 -0400] "GET /manager/ HTTP/1.1" 500 - "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_7) AppleWebKit/534.24 (KHTML, like Gecko) Chrome/11.0.696.68 Safari/534.24""@

I know this doesn't help with reproduction - just wanted to add a little more information to the mix.

**Comment by webandflow** [comment]
webandflow submitted:

What I'm experiencing here may actually be two different issues.... however I able to reproduce the 500 error behavior pretty consistently when:

1) Load up a chunk for editing
2) Saving the chunk (either by command-option-s or via UI button in MODX)
3) Trying to load another chunk for editing as soon as the element list is available again (while "Save Successful" is still visible on the top right

Basically, ff I try to load the next chunk as quickly as I MODX will let me, then I get this 500 error.  Interestingly, though, it doesn't ALWAYS result in the default.cache.php file getting corrupted... so maybe there's another file it affects as well?

**Comment by webandflow** [comment]
webandflow submitted:

Another update... the system that I've been testing on up until now has been at work on a local area network.  I have been trying to replicate the problem on my site5 vps today and haven't been able to.

So, now I'm trying to figure out what about my local environment is different from my site5 environment.

Off the bat, I do know that my local servers use a separate DB server and my site5 vps uses localhost for mysql... so, I'm going to try to run some more tests here locally with an onboard mysql and a dump of the db from the site that is causing me the most trouble right now and see if the problem still arises.

**Comment by webandflow** [comment]
webandflow submitted:

After days of being able to reproduce this at will as described above, for the past couple of days I haven't been able to make it happen once.  I'm beginning to think that it may have something to do with latency on our network - when the conversation between our webservers and our DB servers aren't as quick as it should be for some reason.

In any event, literally without changing anything, I haven't been able to make this happen since Monday.

**Comment by webandflow** [comment]
webandflow submitted:

I guess I spoke too soon.  

I have a fresh install of RC-4 in the same server environment and here's what I'm seeing in the MODX error log

 [2011-05-19 13:49:50](ERROR @ /mnt/glusterfs/www/html/arewedoingthemostgood/core/components/dtmgfeedback/classes/class.dtmgfeedback.php : 16) PHP warning: get_headers() expects at least 1 parameter, 0 given
[2011-05-19 14:07:41](ERROR @ /connectors/system/settings.php) Error writing file: /mnt/glusterfs/www/html/arewedoingthemostgood/core/cache/lexicon_topics/lexicon/en/core/default.cache.php
[2011-05-19 14:07:41](ERROR @ /connectors/system/settings.php) Error caching lexicon topic lexicon/en/core/default
[2011-05-19 14:07:42](ERROR @ /connectors/system/settings.php) Error writing file: /mnt/glusterfs/www/html/arewedoingthemostgood/core/cache/action_map/mgr/actions.cache.php
[2011-05-19 14:07:42](ERROR @ /connectors/system/settings.php) Error writing file: /mnt/glusterfs/www/html/arewedoingthemostgood/core/cache/action_map/mgr/actions.cache.php
[2011-05-19 14:07:42](ERROR @ /connectors/system/settings.php) Error caching action map mgr/actions
[2011-05-19 14:07:42](ERROR @ /connectors/system/settings.php) Error caching action map mgr/actions
[2011-05-19 14:11:38](ERROR @ /index.php) Error writing file: /mnt/glusterfs/www/html/arewedoingthemostgood/core/cache/resource/web/resources/11.cache.php

Checking into permissions/ownership issues, but don't expect that's the problem...

[2011-05-19 14:11:38](ERROR @ /index.php) Could not cache resource 11

**Comment by opengeek** [comment]
opengeek submitted:

I think this must have something to do with the dtmgfeedback component, perhaps? Otherwise, I have no idea how to reproduce this still.

**Comment by webandflow** [comment]
webandflow submitted:

No, nothing to with DTMGFeedback - the first instance didn't have this component (or any custom components).  Unfortunately, I haven't been able to spend much time on these in the past couple of weeks, so I haven't revisited since then, but will keep you posted.
